extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/session/JTASessionOptions.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkus.hibernate.orm.runtime.session;
+
+import org.hibernate.FlushMode;
+import org.hibernate.SessionBuilder;
+import org.hibernate.SessionFactory;
+import org.hibernate.resource.jdbc.spi.PhysicalConnectionHandlingMode;
+
+public class JTASessionOptions {
+    private static final String KEY = JTASessionOptions.class.getName();
+
+    public static void init(SessionFactory sessionFactory) {
+        SessionBuilder<?> options = sessionFactory.withOptions()
+                .autoClose(true) // .owner() is deprecated as well, so it looks like we need to rely on deprecated code...
+                .connectionHandlingMode(
+                        PhysicalConnectionHandlingMode.DELAYED_ACQUISITION_AND_RELEASE_BEFORE_TRANSACTION_COMPLETION)
```

Comment:
Could you explain to me again why these options have to be set again here?

Reply:
Because managed, transaction-scoped sessions have different requirements from user-created sessions (direct calls to `createEntityManager()`) with manual transaction `begin`/`commit`.

Most important is the auto-close on transaction completion, which obviously only makes sense for transaction-scoped sessions. When a user calls `.createEntityManager()` manually, you'd expect them to close the EM manually.

For other settings, it's really about backward compatibility. From what I can see in `FastBootMetadataBuilder` we still allow custom settings for all sorts of properties (from `persistence.xml`?), and while other connection handling modes could make sense for manually-created entity managers, I don't think they would for transaction-scoped sessions.
This solution allows us to force these settings for transaction-scoped sessions, while preserving the ability to override them for manually-created entity managers.

To sum up, we could limit this to only setting `autoClose(true)` here, and assume that people setting the connection handling mode and other related settings manually know what they are doing.